crates/net/downloaders/src/metrics.rs

```diff
@@ -22,6 +22,10 @@ pub struct DownloaderMetrics {
     pub total_flushed: Counter,
     /// Number of items that were successfully downloaded
     pub total_downloaded: Counter,
+    /// The number of out-of-order requests sent by the downloader.
+    /// The consumer of the download stream is able to re-request data (headers or bodies) in case
+    /// it encountered a recoverable error (e.g. during insertion).
+    pub out_of_order_requests: Counter,
```

Comment:
it's not obvious what out-of-order means here exactly.
judging from the docs, these are retries?
should we perhaps rename this to retries?

Reply:
no, these are requests that are sent if the download range has changed:
- for headers, this happens when the sync target tip is different from the previous one.
- for bodies, when the new download range start is less than the last block number returned from the stream.
in both cases, we are redownloading data that we might have downloaded previously. not sure how to make it clear in the docs.

Reply:
I see.
This is not yet really clear in the docs. Mainly because not cleat what `out-of-order` means.

also, why is this metric useful?

Reply:
this is useful to observe and make sure that the "recovery" logic in the downloaders is triggered in rare cases. for headers, this should happen only on a new iteration of the stage loop, for bodies - with current stage error handling this shouldn't be happening at all